### PR TITLE
energy gun infinite cell runtime fix

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
fix: Fixed a runtime when firing energy guns with infinite power cells.
/:cl:

for reviewers, only added `(cell.maxcharge != INFINITY)` to the `if()`, rest is cleaning up redundant code.

### CODE REVIEW

File: code/modules/projectiles/guns/energy.dm [hunk 1]
```diff
@@ -35,21 +35,17 @@
...
+	var/frequency_to_use
 
-	if(shot.e_cost > 0)
-		shot_cost_percent = FLOOR(clamp(shot.e_cost / cell.maxcharge, 0, 1) * 100, 1)
-		max_shots = round(100/shot_cost_percent)
-		shots_left = round(batt_percent/shot_cost_percent)
+	var/obj/item/ammo_casing/energy/shot = ammo_type[select]
+	if((shot.e_cost > 0) && (cell.maxcharge != INFINITY))
+		// What percentage of the full battery a shot will expend
+		var/shot_cost_percent = FLOOR(clamp(shot.e_cost / cell.maxcharge, 0, 1) * 100, 1)
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/energy.dm:44]: `FLOOR(x, 1)` is equivalent to `round(x)` if you want to simplify it

Comment (ShizCalev) [inline code/modules/projectiles/guns/energy.dm:44]:
```suggestion
		var/shot_cost_percent = round(clamp(shot.e_cost / cell.maxcharge, 0, 1) * 100)
```

File: code/modules/projectiles/guns/energy.dm [hunk 2]
```diff
@@ -35,21 +35,17 @@
...
-		max_shots = round(100/shot_cost_percent)
-		shots_left = round(batt_percent/shot_cost_percent)
+	var/obj/item/ammo_casing/energy/shot = ammo_type[select]
+	if((shot.e_cost > 0) && (cell.maxcharge != INFINITY))
+		// What percentage of the full battery a shot will expend
+		var/shot_cost_percent = round(clamp(shot.e_cost / cell.maxcharge, 0, 1) * 100)
+		// The total amount of shots the fully charged energy gun can fire before running out
+		var/max_shots = round(100/shot_cost_percent)
+		// How many shots left before the energy gun's current battery runs out of energy
+		var/shots_left = round((FLOOR(clamp(cell.charge / cell.maxcharge, 0, 1) * 100, 1))/shot_cost_percent)
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/energy.dm:48]: here too

Comment (ShizCalev) [reply code/modules/projectiles/guns/energy.dm:48]:
```suggestion
		var/shots_left = round((round(clamp(cell.charge / cell.maxcharge, 0, 1) * 100))/shot_cost_percent)
```

**Pull request merged**